frame/elections-phragmen/src/lib.rs

```diff
@@ -234,11 +230,17 @@ decl_storage! {
 		/// Votes and locked stake of a particular voter.
 		///
 		/// TWOX-NOTE: SAFE as `AccountId` is a crypto hash
-		pub Voting get(fn voting): map hasher(twox_64_concat) T::AccountId => (BalanceOf<T>, Vec<T::AccountId>);
+		pub Voting get(fn voting): map hasher(twox_64_concat) T::AccountId
+			=> (BalanceOf<T>, Vec<T::AccountId>);
 
 		/// The present candidate list. Sorted based on account-id. A current member or runner-up
 		/// can never enter this vector and is always implicitly assumed to be a candidate.
 		pub Candidates get(fn candidates): Vec<T::AccountId>;
+
+		/// Should be used in conjunction with `on_runtime_upgrade` to ensure an upgrade is executed
+		/// once, even if the code is not removed in time.
+		pub NeedsMigration get(fn needs_migrations): bool = true;
```

Comment:
This does not seem very generic. Should probably follow the Version Enum pattern we have used before

Reply:
Also I think for sanity, it should be `HasMigrated`, where the default value is false, and is set to true when done.

Reply:
Yeah agreed.

Reply:
Moved to enum.